vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/RenderingManager.java

```diff
@@ -0,0 +1,315 @@
+
+                        try {
+                            downloadSequences(choice.getSequence(), module, space, spaceFolderPath, imagesFolderPath);
+                        } catch (FileStorageException e) {
+                            logger.error("Could not download Sequence",e);
+                        } 
+                    }
+                });
+            } else {
+                IContentBlock contentBlock = slide.getFirstImageBlock();
+                if(contentBlock!= null) {
+                    IVSImage image = slide.getFirstImageBlock().getImage();
+                    try {
+                        storageManager.copyImage(image, imagesFolderPath);
+                    } catch (FileStorageException e) {
+                        logger.error("Could not download Sequence",e);
+                    }
+                }
+                try {
+                    renderSlide(slide.getId(), spaceFolderPath ,  space.getId(), module.getId(), startSequence.getId());
+                } catch (FileStorageException e) {
+                    logger.error("Could not store template for the slide", e);
+                }
+            }
+        });
+    }
+
+    /**
+     * 
+     * Stores the processed template for slide into spaceFolderPath
+     * @param slideId                the slide Id 
+     * @param spaceFolderPath        the space folder path where space content will be stored             
+     * @param spaceId                the id of the space to be rendered 
+     * @param moduleId
+     * @param sequenceId
+     * @throws FileStorageException 
+     */
+    private void renderSlide(String slideId, String spaceFolderPath, String spaceId, String moduleId, String sequenceId ) throws FileStorageException {
+        try {      
+            Context thymeleafContext = new Context();
+
+            populateContextForSlide( thymeleafContext, spaceId, moduleId, sequenceId, slideId );
+            String response = springTemplateEngine.process(SLIDE_TEMPLATE_DOWNLOAD_API , thymeleafContext);
+            byte[] fileContent = response.getBytes();
+            storageEngineDownloads.storeFile(fileContent, slideId+DOWNLOAD_FILE_EXTENSION,spaceFolderPath );
+
+        } catch (SlidesInSequenceNotFoundException  | SequenceNotFoundException |SlideNotFoundException e ) {
+            logger.error("Could not add html page for slide" , e);
+        }       
+    }
+
+    
+    /**
+     *@see IRenderingManager#populateContextForSlide(Context, String, String, String, String)
+     */
+    @Override
+    public void populateContextForSlide(Context context, String spaceId, String moduleId, String sequenceId, String slideId) 
+            throws SlidesInSequenceNotFoundException, SequenceNotFoundException, SlideNotFoundException {
+        IModule module = moduleManager.getModule(moduleId);
+        context.setVariable("module", module);
+        String startSequenceId = module.getStartSequence().getId();
+
```

Comment:
why is here an empty line? line 243 belongs logically to line 245, not line 242.